lighthouse-core/gather/driver.js

```diff
@@ -264,25 +264,18 @@ class Driver {
   }
 
   getAppManifest() {
-    return new Promise((resolve, reject) => {
-      this.sendCommand('Page.getAppManifest')
-        .then(response => {
-          // We're not reading `response.errors` however it may contain critical and noncritical
-          // errors from Blink's manifest parser:
-          //   https://chromedevtools.github.io/debugger-protocol-viewer/tot/Page/#type-AppManifestError
-          if (!response.data) {
-            if (response.url) {
-              return reject(new Error(`Unable to retrieve manifest at ${response.url}.`));
-            }
-
-            // If both the data and the url are empty strings, the page had no manifest.
-            return reject('No web app manifest found.');
-          }
+    return this.sendCommand('Page.getAppManifest')
+      .then(response => {
+        // We're not reading `response.errors` however it may contain critical and noncritical
+        // errors from Blink's manifest parser:
+        //   https://chromedevtools.github.io/debugger-protocol-viewer/tot/Page/#type-AppManifestError
+        if (!response.data) {
```

Comment:
> "Unable to retrieve manifest" is a bad error string for that and we should be as specific as possible to help with those situations

One downside of the current `Page.getAppManifest` protocol call is that it gives no network status information (as far as I can tell). So e.g. if the server returns a 404 with a `Cannot GET /manifest.json` response string, the protocol will just return a `JSON.parse` error from trying to parse that string and we just have to infer that there was a 404. Maybe we could see about getting more detail there.